### DISCUSSION TITLE
Gyro support or "Only M&K" button

### DISCUSSION THREAD

**Discussion by avelixo** [ChristopherKlay/StadiaEnhanced#71] [Ideas & Requests]
I have a suggestion. 

Is it possible that you can include support for  gyroscope (DualSense, DualShock, Switch Pro Controller and Steam Controller)? You could use the ["JoyShockMapper" project](https://github.com/JibbSmart/JoyShockMapper) (which is open source and DRM-free).

If not, could you add a button that only allows mouse and keyboard inputs (although the controllers are connected to the PC)?

My idea is to use JoyShockMapper to convert controller's inputs to mouse/keyboard inputs and only send this (not controller's inputs). 

The problem is that Stadia receives both inputs. The solution would be to disable the controller inputs.

**Comment by ChristopherKlay** [comment]
Sorry for the late reply!

I'm definitely looking into the controller side as a whole, not just because of feature support but also possible issues (like the inability to control the UI when scrollbars are active). Sadly my knowledge about the gamepad API is limited to inputs/processing and other tasks, so i have to currently learn what's actually possible (i.e. disabling registered gamepads temporarily).

I'm definitely working on it however (:

**Reply by avelixo** [reply]
Don't worry about that. In any case, I am the one who thanks you for taking your time to answer me. ;)

I found a solution in a Chrome extension. It's called "ControlStadia" (https://chrome.google.com/webstore/detail/controlstadia/faodiopjdcjfeflfmeadfkgodcbioejp) and his project is in GitHub (https://github.com/ihatecsv/ControlStadia) with GPL 3.0 license but it's not updated since 10 months ago (I think it's abandoned).

"ControlStadia" allows mapping up to four physical gamepads to a single "standard" virtual gamepad which works with Stadia. So, when I connect my gyro controller, I set all the mapping in "[unset]".  Then, I run JoyShockMapper.exe (https://github.com/JibbSmart/JoyShockMapper/releases) locally. And this is how the gyroscope in Stadia works. :D

Although I have two problems with this: 
- I notice a little bit of latency (the author already warned about this and therefore planned to improve it).
- there is no vibration for the controller (I think this is impossible to solve without official gyro support because the gyro controller (emulated) "is" virtually a mouse and keyboard).

Could you integrate this solution into your extension?
